### ISSUE TITLE
Custom binary location fails on macOS

### ISSUE THREAD

**Issue by ciscoo** [SpectoLabs/hoverfly-java#238]
## Description of the bug

Related to the feature I requested a few weeks ago: https://github.com/SpectoLabs/hoverfly-java/issues/236

A colleague of mine was facing the same issue I was which reminded me of the above issue. I went to verify the changes using 0.13.1 and custom binary location worked as I could see the directory and binary being created, but when it came to execution of the Hoverfly binary, it fails on macOS.

My colleague uses a Windows machine so he did not face this issue. He verified that the test was also failing on our internal Jenkins instance as well. 

I've verified the behavior on both Windows and macOS; most likely fails on Linux distributions as well.

## Steps to reproduce the issue

On a unix/linux machine:

1. Generate a Spring Boot Gradle project from https://start.spring.io/
    * You can use [this](https://start.spring.io/#!type=gradle-project&language=java&platformVersion=2.3.2.RELEASE&packaging=jar&jvmVersion=1.8&groupId=com.example&artifactId=demo&name=demo&description=Demo%20project%20for%20Spring%20Boot&packageName=com.example.demo&dependencies=web) link for the project I generated
2. Include `testImplementation 'io.specto:hoverfly-java-junit5:0.13.1'`
3. Write a test like:
```java
@HoverflySimulate(
		source = @HoverflySimulate.Source(value = "simulation.json", type = HoverflySimulate.SourceType.CLASSPATH),
		config = @HoverflyConfig(binaryLocation = "hoverfly-tmp")
)
@ExtendWith(HoverflyExtension.class)
@SpringBootTest
class DemoApplicationTests {

	@Test
	void contextLoads() {
	}

}
```
5. Run test and watch it fail

### Observed result

**Hoverfly Java and Hoverfly error messages seen (If none, say none)**

```stacktrace
java.lang.IllegalStateException: Could not start Hoverfly process

	at io.specto.hoverfly.junit.core.Hoverfly.startHoverflyProcess(Hoverfly.java:259)
	at io.specto.hoverfly.junit.core.Hoverfly.start(Hoverfly.java:140)
	at io.specto.hoverfly.junit5.HoverflyExtension.beforeAll(HoverflyExtension.java:114)
	--- snip
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
Caused by: org.zeroturnaround.exec.ProcessInitException: Could not execute [hoverfly-tmp/hoverfly_OSX_amd64, -pp, 56484, -ap, 56485, -logs, json, -log-level, info] in hoverfly-tmp. Error=2, No such file or directory
	at org.zeroturnaround.exec.ProcessInitException.newInstance(ProcessInitException.java:80)
	at org.zeroturnaround.exec.ProcessExecutor.invokeStart(ProcessExecutor.java:1022)
	at org.zeroturnaround.exec.ProcessExecutor.startInternal(ProcessExecutor.java:989)
	at org.zeroturnaround.exec.ProcessExecutor.start(ProcessExecutor.java:961)
	at io.specto.hoverfly.junit.core.Hoverfly.startHoverflyProcess(Hoverfly.java:257)
	... 37 more
Caused by: java.io.IOException: Cannot run program "hoverfly-tmp/hoverfly_OSX_amd64" (in directory "hoverfly-tmp"): error=2, No such file or directory
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at org.zeroturnaround.exec.ProcessExecutor.invokeStart(ProcessExecutor.java:1016)
	... 40 more
Caused by: java.io.IOException: error=2, No such file or directory
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:319)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:250)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 42 more
```

Test passes successfully on Windows:

<img width="1308" src="https://user-images.githubusercontent.com/6423628/88806905-a868ad80-d176-11ea-827f-2948533fd187.png">

Test fails on macOS:
 
<img width="1308" alt="Screen Shot 2020-07-29 at 8 24 42 AM" src="https://user-images.githubusercontent.com/6423628/88806943-b3234280-d176-11ea-8109-c6c30d7851bd.png">

### Expected result

Sample test above should pass regardless of OS.

## Additional relevant information

1. Hoverfly Java version: `io.specto:hoverfly-java-junit5:0.13.1`

2. Anything that might help us to diagnose the problem:

I think this line:

https://github.com/SpectoLabs/hoverfly-java/blob/503ade151387d132af6ea065d048a7487b3be8f5/src/main/java/io/specto/hoverfly/junit/core/Hoverfly.java#L256

should be:

```diff
- .directory(tempFileManager.getTempDirectory().toFile())
+ .directory(tempFileManager.getTempDirectory().toAbsolutePath().toFile())
```

I observed from debugging that the path resolved here is something like `hoverfly-tmp/amd_xxxx`, but my assumption is the "context" of _where_ the process is launched changes the current working directory or something. This doesn't seem to affect Windows, but a fully qualified path may help for unix/linux machines.

**Comment by tommysitu** [comment]
Thank you for reporting the issue! I think what you suggest would work. I can raise a PR.

**Comment by tommysitu** [comment]
Hi @ciscoo this has been fixed in `0.13.2-SNAPSHOT`

**Comment by ciscoo** [comment]
@tommysitu confirmed working. 🎉 

Before:

```logs
$ ./gradlew clean build

> Task :test FAILED

DemoApplicationTests > initializationError FAILED
    java.lang.IllegalStateException at Hoverfly.java:259
        Caused by: org.zeroturnaround.exec.ProcessInitException at ProcessInitException.java:80
            Caused by: java.io.IOException at ProcessBuilder.java:1128
                Caused by: java.io.IOException at ProcessImpl.java:-2

1 test completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///Users/cisco/Desktop/demo/build/reports/tests/test/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
10 actionable tasks: 10 executed
```

After:

```logs
$ ./gradlew clean build

> Task :test
2020-08-03 12:58:06.717  INFO 7674 --- [extShutdownHook] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'

BUILD SUCCESSFUL in 7s
10 actionable tasks: 10 executed
```

Full intellij logs:

```logs
13:03:49.543 [main] INFO io.specto.hoverfly.junit.core.TempFileManager - Selecting the following binary based on the current operating system: hoverfly_OSX_amd64
13:03:49.553 [main] INFO io.specto.hoverfly.junit.core.TempFileManager - Storing binary in temporary directory /Users/cisco/Desktop/demo/hoverfly-tmp/hoverfly_OSX_amd64
13:03:49.900 [main] INFO io.specto.hoverfly.junit.core.Hoverfly - Executing binary at /Users/cisco/Desktop/demo/hoverfly-tmp/hoverfly_OSX_amd64
13:03:49.921 [main] DEBUG org.zeroturnaround.exec.ProcessExecutor - Executing [/Users/cisco/Desktop/demo/hoverfly-tmp/hoverfly_OSX_amd64, -pp, 54570, -ap, 54571, -logs, json, -log-level, info] in /Users/cisco/Desktop/demo/hoverfly-tmp.
13:03:49.979 [main] DEBUG org.zeroturnaround.exec.ProcessExecutor - Started Process[pid=8294, exitValue="not exited"]
13:03:50.127 [main] DEBUG io.specto.hoverfly.junit.api.HoverflyClient - Hoverfly healthcheck failed: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:54571
13:03:50.232 [main] DEBUG io.specto.hoverfly.junit.api.HoverflyClient - Hoverfly healthcheck failed: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:54571
13:03:50.337 [main] DEBUG io.specto.hoverfly.junit.api.HoverflyClient - Hoverfly healthcheck failed: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:54571
13:03:50.412 [Thread-1] INFO hoverfly - Default proxy port has been overwritten port=54570
13:03:50.413 [Thread-1] INFO hoverfly - Default admin port has been overwritten port=54571
13:03:50.414 [Thread-1] INFO hoverfly - Using memory backend 
13:03:50.414 [Thread-1] INFO hoverfly - Proxy prepared... Destination=. Mode=simulate ProxyPort=54570
13:03:50.415 [Thread-1] INFO hoverfly - current proxy configuration destination=. mode=simulate port=54570
13:03:50.415 [Thread-1] INFO hoverfly - serving proxy 
13:03:50.416 [Thread-1] INFO hoverfly - Admin interface is starting... AdminPort=54571
13:03:50.492 [Thread-1] INFO hoverfly - Mode has been changed mode=simulate
13:03:50.511 [main] INFO io.specto.hoverfly.junit.core.ProxyConfigurer - Setting proxy host to localhost
13:03:50.511 [main] INFO io.specto.hoverfly.junit.core.ProxyConfigurer - Setting proxy proxyPort to 54570
13:03:50.511 [main] INFO io.specto.hoverfly.junit.core.Hoverfly - Importing simulation data to Hoverfly
13:03:50.600 [main] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating CacheAwareContextLoaderDelegate from class [org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate]
13:03:50.635 [main] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating BootstrapContext using constructor [public org.springframework.test.context.support.DefaultBootstrapContext(java.lang.Class,org.springframework.test.context.CacheAwareContextLoaderDelegate)]
13:03:50.727 [main] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating TestContextBootstrapper for test class [com.example.demo.DemoApplicationTests] from class [org.springframework.boot.test.context.SpringBootTestContextBootstrapper]
13:03:50.790 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Neither @ContextConfiguration nor @ContextHierarchy found for test class [com.example.demo.DemoApplicationTests], using SpringBootContextLoader
13:03:50.800 [main] DEBUG org.springframework.test.context.support.AbstractContextLoader - Did not detect default resource location for test class [com.example.demo.DemoApplicationTests]: class path resource [com/example/demo/DemoApplicationTests-context.xml] does not exist
13:03:50.804 [main] DEBUG org.springframework.test.context.support.AbstractContextLoader - Did not detect default resource location for test class [com.example.demo.DemoApplicationTests]: class path resource [com/example/demo/DemoApplicationTestsContext.groovy] does not exist
13:03:50.830 [main] INFO org.springframework.test.context.support.AbstractContextLoader - Could not detect default resource locations for test class [com.example.demo.DemoApplicationTests]: no resource found for suffixes {-context.xml, Context.groovy}.
13:03:50.841 [main] INFO org.springframework.test.context.support.AnnotationConfigContextLoaderUtils - Could not detect default configuration classes for test class [com.example.demo.DemoApplicationTests]: DemoApplicationTests does not declare any static, non-private, non-final, nested classes annotated with @Configuration.
13:03:50.997 [main] DEBUG org.springframework.test.context.support.ActiveProfilesUtils - Could not find an 'annotation declaring class' for annotation type [org.springframework.test.context.ActiveProfiles] and class [com.example.demo.DemoApplicationTests]
13:03:51.348 [main] DEBUG org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider - Identified candidate component class: file [/Users/cisco/Desktop/demo/out/production/classes/com/example/demo/DemoApplication.class]
13:03:51.352 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Found @SpringBootConfiguration com.example.demo.DemoApplication for test class com.example.demo.DemoApplicationTests
13:03:51.612 [main] DEBUG org.springframework.boot.test.context.SpringBootTestContextBootstrapper - @TestExecutionListeners is not present for class [com.example.demo.DemoApplicationTests]: using defaults.
13:03:51.615 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Loaded default TestExecutionListener class names from location [META-INF/spring.factories]: [org.springframework.security.test.context.support.WithSecurityContextTestExecutionListener, org.springframework.security.test.context.support.ReactorContextTestExecutionListener, org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener, org.springframework.boot.test.autoconfigure.webservices.client.MockWebServiceServerTestExecutionListener, org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener, org.springframework.test.context.web.ServletTestExecutionListener, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener, org.springframework.test.context.support.DependencyInjectionTestExecutionListener, org.springframework.test.context.support.DirtiesContextTestExecutionListener, org.springframework.test.context.transaction.TransactionalTestExecutionListener, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener, org.springframework.test.context.event.EventPublishingTestExecutionListener]
13:03:51.645 [main] DEBUG org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Skipping candidate TestExecutionListener [org.springframework.test.context.transaction.TransactionalTestExecutionListener] due to a missing dependency. Specify custom listener classes or make the default listener classes and their required dependencies available. Offending class: [org/springframework/transaction/interceptor/TransactionAttributeSource]
13:03:51.646 [main] DEBUG org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Skipping candidate TestExecutionListener [org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener] due to a missing dependency. Specify custom listener classes or make the default listener classes and their required dependencies available. Offending class: [org/springframework/transaction/interceptor/TransactionAttribute]
13:03:51.646 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Using TestExecutionListeners: [org.springframework.test.context.web.ServletTestExecutionListener@2575f671, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener@329a1243, org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener@ecf9fb3, org.springframework.boot.test.autoconfigure.SpringBootDependencyInjectionTestExecutionListener@2d35442b, org.springframework.test.context.support.DirtiesContextTestExecutionListener@27f9e982, org.springframework.security.test.context.support.WithSecurityContextTestExecutionListener@4593ff34, org.springframework.test.context.event.EventPublishingTestExecutionListener@37d3d232, org.springframework.security.test.context.support.ReactorContextTestExecutionListener@30c0ccff, org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener@581d969c, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener@22db8f4, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener@2b46a8c1, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener@1d572e62, org.springframework.boot.test.autoconfigure.webservices.client.MockWebServiceServerTestExecutionListener@29caf222, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener@46cf05f7]
13:03:51.651 [main] DEBUG org.springframework.test.context.support.AbstractDirtiesContextTestExecutionListener - Before test class: context [DefaultTestContext@2a2bb0eb testClass = DemoApplicationTests, testInstance = [null], testMethod = [null], testException = [null], mergedContextConfiguration = [WebMergedContextConfiguration@3c291aad testClass = DemoApplicationTests, locations = '{}', classes = '{class com.example.demo.DemoApplication}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{}', propertySourceProperties = '{org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true}', contextCustomizers = set[org.springframework.boot.test.autoconfigure.properties.PropertyMappingContextCustomizer@0, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverContextCustomizerFactory$Customizer@51cd7ffc, org.springframework.boot.test.context.filter.ExcludeFilterContextCustomizer@6b9ce1bf, org.springframework.boot.test.json.DuplicateJsonObjectContextCustomizerFactory$DuplicateJsonObjectContextCustomizer@4fc5e095, org.springframework.boot.test.mock.mockito.MockitoContextCustomizer@0, org.springframework.boot.test.web.client.TestRestTemplateContextCustomizer@5b40ceb, org.springframework.boot.test.context.SpringBootTestArgs@1], resourceBasePath = 'src/main/webapp', contextLoader = 'org.springframework.boot.test.context.SpringBootContextLoader', parent = [null]], attributes = map['org.springframework.test.context.web.ServletTestExecutionListener.activateListener' -> true]], class annotated with @DirtiesContext [false] with mode [null].
13:03:51.713 [main] DEBUG org.springframework.test.context.support.TestPropertySourceUtils - Adding inlined properties to environment: {spring.jmx.enabled=false, org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true}

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v2.3.2.RELEASE)

2020-08-03 13:03:52.440  INFO 8293 --- [           main] com.example.demo.DemoApplicationTests    : Starting DemoApplicationTests on pi with PID 8293 (started by cisco in /Users/cisco/Desktop/demo)
2020-08-03 13:03:52.442  INFO 8293 --- [           main] com.example.demo.DemoApplicationTests    : No active profile set, falling back to default profiles: default
2020-08-03 13:03:54.443  INFO 8293 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2020-08-03 13:03:54.901  INFO 8293 --- [           main] .s.s.UserDetailsServiceAutoConfiguration : 

Using generated security password: 6047abcf-458e-4096-9610-c61951819da1

2020-08-03 13:03:55.062  INFO 8293 --- [           main] o.s.s.web.DefaultSecurityFilterChain     : Creating filter chain: any request, [org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter@385d819, org.springframework.security.web.context.SecurityContextPersistenceFilter@1b4872bc, org.springframework.security.web.header.HeaderWriterFilter@531ec978, org.springframework.security.web.authentication.logout.LogoutFilter@61e86192, org.springframework.security.web.savedrequest.RequestCacheAwareFilter@4dfdfe7d, org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter@364b1061, org.springframework.security.web.authentication.AnonymousAuthenticationFilter@4315c28c, org.springframework.security.web.session.SessionManagementFilter@57f8951a, org.springframework.security.web.access.ExceptionTranslationFilter@5ac53c06]
2020-08-03 13:03:55.228  INFO 8293 --- [           main] com.example.demo.DemoApplicationTests    : Started DemoApplicationTests in 3.486 seconds (JVM running for 8.629)

2020-08-03 13:03:55.263  INFO 8293 --- [           main] io.specto.hoverfly.junit.core.Hoverfly   : Importing simulation data to Hoverfly
2020-08-03 13:03:55.263  INFO 8293 --- [       Thread-1] hoverfly                                 : Mode has been changed mode=simulate

2020-08-03 13:03:55.962  INFO 8293 --- [           main] io.specto.hoverfly.junit.core.Hoverfly   : Destroying hoverfly process
2020-08-03 13:03:55.988  INFO 8293 --- [extShutdownHook] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'

Process finished with exit code 0
```
